### PR TITLE
Fix typos in the architecture README

### PR DESCRIPTION
**Pull request by black-adder**

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -20,26 +20,25 @@ A **Trace** is a data/execution path through the system, and can be thought of a
 This section details the constituents of Jaeger, and how they relate to each other. It is arranged by the order in which spans from your application interact with them. 
 
 ###Jaeger client
```
Comment (yurishkuro) [inline docs/architecture.md]: clients (plural)

Comment (yurishkuro) [inline docs/architecture.md]: or even better, client libraries

File: docs/architecture.md [hunk 2]
```diff
@@ -20,26 +20,25 @@ A **Trace** is a data/execution path through the system, and can be thought of a
...
-Jaeger clients are language specific implementations of the OpenTracing API. The contain a Tracer, which is an interface for Span creation, and context propagation. 
+Jaeger clients are language specific implementations of the OpenTracing API. They contain a Tracer, which is an interface for Span creation, and context propagation. 
 Some common frameworks (e.g. dropwizard) are instrumented out of the box, making it easy to get started. 
 
-An instrumented library creates spans when receiving new requests, and attaches context information (trace id, and span id) to outgoing requests. Note that only ids are propagated with requests, and any other information like operation name, logs, etc is not propagated, and is simply collected at the service by the Jaeger client.
+An instrumented library creates spans when receiving new requests, and attaches context information (trace id, span id, and baggage) to outgoing requests. Note that only ids and baggage are propagated with requests, and any other information like operation name, logs, etc is not propagated, and is simply collected at the service by the Jaeger client.
```
Comment (yurishkuro) [inline docs/architecture.md]: s/Note that only ids/Only ids/

Comment (yurishkuro) [inline docs/architecture.md]: * s/and any other/all other/
 * etc. (period)
 * s/and is simply collected at the service by the Jaeger client./it is transmitted out of process asynchronously, in the background, to Jaeger Agents./

Comment (yurishkuro) [inline docs/architecture.md]: * s/and is simply collected at the service by the Jaeger client./it is transmitted out of process asynchronously, in the background, to Jaeger Agents, and only for sampled traces./

File: docs/architecture.md [hunk 3]
```diff
@@ -17,29 +17,28 @@ A **Trace** is a data/execution path through the system, and can be thought of a
 ![Architecture](images/architecture.png)
 *The current Jaeger architecture*
 
-This section details the constituents of Jaeger, and how they relate to each other. It is arranged by the order in which spans from your application interact with them. 
+This section details the constituents of Jaeger and how they relate to each other. It is arranged by the order in which spans from your application interact with them. 
 
-###Jaeger client
-Jaeger clients are language specific implementations of the OpenTracing API. The contain a Tracer, which is an interface for Span creation, and context propagation. 
+###Jaeger client libraries
+Jaeger clients are language specific implementations of the OpenTracing API. They contain a Tracer, which is an interface for Span creation, and a mechanism for context propagation.
```
Comment (black-adder) [inline docs/architecture.md]: and a mechanism for context propagation is a lie.

How about, "a mechanism to extract tracing specific metadata from propagated context"?

Comment (yurishkuro) [inline docs/architecture.md]: I'd drop that sentence altogether. How about this instead

> They can be used to manually instrument applications for distributed tracing, or with a variety of existing open source frameworks, such as Flask, Dropwizard, gRPC, and many more, that are already integrated with OpenTracing.

File: docs/architecture.md [hunk 4]
```diff
@@ -17,29 +17,28 @@ A **Trace** is a data/execution path through the system, and can be thought of a
 ![Architecture](images/architecture.png)
 *The current Jaeger architecture*
 
-This section details the constituents of Jaeger, and how they relate to each other. It is arranged by the order in which spans from your application interact with them. 
+This section details the constituents of Jaeger and how they relate to each other. It is arranged by the order in which spans from your application interact with them. 
 
-###Jaeger client
-Jaeger clients are language specific implementations of the OpenTracing API. The contain a Tracer, which is an interface for Span creation, and context propagation. 
+###Jaeger client libraries
+Jaeger clients are language specific implementations of the OpenTracing API. They can be used to manually instrument applications for distributed tracing, or with a variety of existing open source frameworks, such as Flask, Dropwizard, gRPC, and many more, that are already integrated with OpenTracing.
```
Comment (yurishkuro) [inline docs/architecture.md]: Sorry, minor fix, move "manually" next to comma

They can be used to instrument applications for distributed tracing *either manually* or with a variety

**Pull request merged**